### PR TITLE
Copy set of receivers in case it changes under us

### PR DESCRIPTION
**Pull request by meejah**
See ticket #578: if an embedded component raises an exception
from a subsciption callback, the connection is removed -- causing
the observations to be dropped. This is synchronous as the
ITransport is RouterApplicationSession.

### CODE REVIEW

**Comment by oberstet** [comment]
Copying will remove this issue, but will produce more GC pressure. Imagine a set of 100k subscribers to a topic. Each publish will copy the set. Do you see a way to "fix it" without copying?

**Comment by meejah** [comment]
Yes, either of the other approaches mentioned in #578 will work: a ``reactor.callLater(0, ...)`` on either the ``.detach()`` or upstream ``onMessage`` or ``.send()`` -- but then that adds to the "things in reactor queue" instead of "copies of the receiver sets"... :/

**Comment by meejah** [comment]
I guess the "least widespread consequences" would be to simply wrap the ``.detach()`` in a ``reactor.callLater(..)`` (here: https://github.com/crossbario/crossbar/blob/master/crossbar/router/session.py#L133). (This does work, at least for my test-case for the original problem).

**Comment by meejah** [comment]
Another alternative would be to store the observers as a ``list()`` instead of a set since you can iterate a list and (safely) change the size of it while iterating. But, then we have to keep uniqueness etc... :/

**Comment by oberstet** [comment]
@meejah So I close this one, as I merged your other PR https://github.com/crossbario/crossbar/pull/668 - which does not carry the overhead that this PR would have had.
